src/app/components/Persist/UnlockForm.tsx

```diff
@@ -0,0 +1,82 @@
+import { Box, Button, Form, Layer, Paragraph } from 'grommet'
+import { persistActions } from 'app/state/persist'
+import { selectEnteredWrongPassword } from 'app/state/persist/selectors'
+import React from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import { useTranslation } from 'react-i18next'
+import { PasswordField } from 'app/components/PasswordField'
+import { Header } from 'app/components/Header'
+import { useNavigate } from 'react-router-dom'
+
+export function UnlockForm() {
+  const { t } = useTranslation()
+  const dispatch = useDispatch()
+  const navigate = useNavigate()
+  const enteredWrongPassword = useSelector(selectEnteredWrongPassword)
+  const [password, setPassword] = React.useState('')
+
+  return (
+    <Layer modal background="background-front">
+      <Box pad="medium" gap="medium" direction="row" align="center" responsive={false}>
+        <Form>
+          <Header>{t('persist.loginToProfile.title', 'Welcome Back!')}</Header>
+          <Paragraph>
+            <label htmlFor="password">
+              {t(
+                'persist.loginToProfile.description',
+                'Log into your existing user profile on this computer to access the wallets you already added.',
+              )}
+            </label>
+          </Paragraph>
+
+          <PasswordField
+            placeholder={t('persist.loginToProfile.enterPasswordHere', 'Enter your password here')}
+            name="password"
+            inputElementId="password"
+            autoComplete="current-password"
+            autoFocus
+            value={password}
+            onChange={event => setPassword(event.target.value)}
+            error={enteredWrongPassword ? t('persist.loginToProfile.wrongPassword', 'Wrong password') : false}
+            showTip={t('persist.loginToProfile.showPassword', 'Show password')}
+            hideTip={t('persist.loginToProfile.hidePassword', 'Hide password')}
+            width="auto"
+          ></PasswordField>
+
+          <Box direction="row" justify="between" margin={{ top: 'medium' }}>
+            <Button
+              type="submit"
+              label={t('persist.loginToProfile.unlock', 'Unlock')}
+              primary
+              onClick={() => dispatch(persistActions.unlockAsync({ password: password }))}
+            />
+
+            <Button
+              label={t('persist.loginToProfile.skipUnlocking', 'Continue without the profile')}
+              onClick={() => {
+                navigate('/')
```

Comment:
Good UX question. I've been considering the skipped profile like incognito mode, not read-only profile.
- Against:
  - We've removed search for foreign accounts, so read-only view is barely supported.
  - I was considering removing remove account ID from URL. To have fewer sources of state. That would also increase privacy - after locking wallet, someone couldn't check browser history (or click skip) to get your address.
- For:
  - Would allow wallet owner to quickly get **their account's address**. Pre-redirect URL might not be locked account (open a foreign account URL from bookmark and click skip). More complex proposal is to have a wallet pin to unlock read-only, and only ask for password before signing transactions.
  - Would allow quickly viewing **foreign account's balance** from URL/bookmark. But this is just worse search.